settings/src/main/resources/client-settings.schema.json

```diff
@@ -315,10 +312,13 @@
         "save_mid": {
           "type": "boolean"
         },
-        "save_as": {
+        "mark_as_finalized": {
```

Comment:
Should `mark_as_finalized` be removed?

Reply:
Done.

Reply:
No, if we remove it scanning old QR codes won't work as expected because this option will not be recognized and will be skipped.

Reply:
Ah, so we need the schema to recognize the old fields so that we can migrate them?

Reply:
Correct. I've added a new test for this: https://github.com/getodk/collect/pull/5556/commits/0772f89b92e6debe5e913da5c0ec4ba43c505dac